backends/gstreamer/lib.rs

```diff
@@ -152,3 +187,246 @@ impl<'a> Drop for MutexGuard<'a> {
         }
     }
 }
+
+struct GStreamerRegistryScanner {
+    audio_decoder_factories: Vec<gst::ElementFactory>,
+    audio_parser_factories: Vec<gst::ElementFactory>,
+    video_decoder_factories: Vec<gst::ElementFactory>,
+    video_parser_factories: Vec<gst::ElementFactory>,
+    demux_factories: Vec<gst::ElementFactory>,
```

Comment:
Why do we need `*_factories` to be members of `GStreamerRegistryScanner`? It seems that we could have them as local variables for the `initialize` method.